### PR TITLE
Branches -- Eve & Paige

### PR DESCRIPTION
**Pull request by minipaige02**
# Rideshare-Rails
Congratulations! You're submitting your assignment! These comprehension questions should be answered by both partners together, not by a single teammate.

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe the types of entity relationships you set up in your project and why you set up the relationships that way | A passenger can have many trips, a driver can have many trips, and a trip belongs to a driver and a passenger. We set them up that way because it made the most sense in the scope of this project. We also allowed for nullify on dependency so that we could delete trips, passengers, or divers. 
Describe the role of model validations in your application | Validations in the model verify input so that an instance of the model cannot be created without ensure the data is correct. This helps our data in our database stay clean!
How did your team break up the work to be done? | We did some work together at Ada and then broke up other tasks to complete on our own at home. We sometimes broke things up by our preferences, i.e. if one of us like to do more backend vs. front end. We also practiced branching and code review via GitHub.
What features did you choose to prioritize in your project, and what features, if any, did you have to set aside to meet the deadline? | We prioritized the CRUD features for the Driver and Passenger first and then went on to Trip and then on to more "business tasks".
What was one thing that your team collectively gained more clarity on after completing this assignment? | We gained more clarity on the way models work with a database and how the controller functions in an application, and how it interacts with a browser. We also gained more clarity on branching and merging with git.
What is your Trello board URL? | https://trello.com/b/ug7GjFQt/rideshare-rails
What is the Heroku URL of your deployed application? | https://tofus-ride-share-app.herokuapp.com/
What are two discussion points that you and your pair discussed when giving/receiving feedback from each other that you would be willing to share? | We discussed both of our preferences for being kind when giving feedback and speaking up when we weren't clear on something or had ideas for other ways to do things.

### CODE REVIEW

File: app/controllers/trips_controller.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    if @trip.nil?
+      head :not_found
+      return
+    end
+  end
+  
+  def new
+    passenger_id = params[:passenger_id]
+    @passenger = Passenger.find_by(id: passenger_id)
+    @trip = Trip.new(@passenger.request_trip[:trip])
```
Comment (kaidamasaki) [inline app/controllers/trips_controller.rb:15]: This causes an error if there are no free drivers.

**Comment by kaidamasaki** [comment]
# Rideshare Rails
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate git usage with no extraneous files checked in and all team members contributing | 👍🏼
Answered comprehension questions | 👍🏼
Uses named routes (like `_path`) | 👍🏼
RESTful routes utilized | 👍🏼
**Project Requirements** |
Table relationships | 👍🏼
Validation rules for Models | 👍🏼
Business logic is in the models | 👍🏼
Tests for validations | 👍🏼
Tests for relationships | Forgot to test `Trip`'s `belongs_to`s.
Tests for custom model methods | 👍🏼
Tests for controller actions | 👍🏼
Database is seeded from the CSV files | 👍🏼
Trello board is created and utilized in project management | 👍🏼
Heroku instance is online | 👍🏼
The app is styled to create an attractive user interface | 👍🏼
**Overall** | Well done!  Your site was clean and attractive and I liked your theme.  You had one issue where if there are no free drivers you get an error; remember to test weird edge cases like that in the future. Other than that everything looked great and was well tested.
